### PR TITLE
Move OpenCL sources from intel/llvm to UR 

### PR DESCRIPTION
**Pull request by fabiomestre**

### CODE REVIEW

**Review by kbenzie — COMMENTED** [review]
Please add this to the `.github/CODEOWNERS` file:

```
# OpenCL adapter
source/adapters/opencl          @oneapi-src/unified-runtime-opencl-write
```

**Review by veselypeta — APPROVED** [review]
LGTM

File: test/conformance/CMakeLists.txt
Comment (kbenzie) [inline test/conformance/CMakeLists.txt:56]:
```suggestion
        if(UR_BUILD_ADAPTER_OPENCL)
```

**Review by kbenzie — APPROVED** [review]
1 minor thing

**Comment by pbalcer** [comment]
How do we plan on testing this adapter? using a CPU runtime (would it be possible to do that on GH-hosted runner)? something else?
Since this patch isn't adding a CTS workflow, there should at the very least be an issue to create one.

**Review by martygrant — APPROVED** [review]

**Comment by kbenzie** [comment]
> How do we plan on testing this adapter? using a CPU runtime (would it be possible to do that on GH-hosted runner)? something else?

These are the options:

* Intel OpenCL CPU - could this be used on GitHub hosted runners?
* Intel OpenCL GPU on the Intel ARC runner - already a constrained resource
* Intel OpenCL GPU on the Intel iGPU on the AMD GPU runner - already a constrained resource
* Use [codeplaysoftware/oneapi-construction-kit](https://github.com/codeplaysoftware/oneapi-construction-kit) - can be used on GitHub hosted runners but full DPC++ support is still in progress
* Maybe some combination of the above to load balance resources?

> Since this patch isn't adding a build/CTS workflow, there should at the very least be an issue to create one.

Agreed. @fabiomestre could you create that please?

**Comment by pbalcer** [comment]
> * Intel OpenCL CPU - could this be used on GitHub hosted runners?

Let's start with this one. Hopefully it's straightforward to do.

> * Intel OpenCL GPU on the Intel ARC runner - already a constrained resource
> * Intel OpenCL GPU on the Intel iGPU on the AMD GPU runner - already a constrained resource
> * Use [codeplaysoftware/oneapi-construction-kit](https://github.com/codeplaysoftware/oneapi-construction-kit) - can be used on GitHub hosted runners but full DPC++ support is still in progress
> * Maybe some combination of the above to load balance resources?

I wouldn't want to add too many additional jobs on the existing runners, the queue times are already significant. I'm hoping we can either add more platforms or be more efficient with what we have through virtualization. I'm going to ask the team to prioritize adding a new intel-gpu runner in the short-term.

**Comment by fabiomestre** [comment]
> > How do we plan on testing this adapter? using a CPU runtime (would it be possible to do that on GH-hosted runner)? something else?
> 
> These are the options:
> 
> * Intel OpenCL CPU - could this be used on GitHub hosted runners?
> * Intel OpenCL GPU on the Intel ARC runner - already a constrained resource
> * Intel OpenCL GPU on the Intel iGPU on the AMD GPU runner - already a constrained resource
> * Use [codeplaysoftware/oneapi-construction-kit](https://github.com/codeplaysoftware/oneapi-construction-kit) - can be used on GitHub hosted runners but full DPC++ support is still in progress
> * Maybe some combination of the above to load balance resources?
> 
> > Since this patch isn't adding a build/CTS workflow, there should at the very least be an issue to create one.
> 
> Agreed. @fabiomestre could you create that please?

Created this issue to track the task: https://github.com/oneapi-src/unified-runtime/issues/964

**Review by pbalcer — APPROVED** [review]

**Comment by pbalcer** [comment]
There's no point in waiting for CI because the runners are out of commission today, and the adapter this PR is adding isn't built by any worker anyway. So I think we should just merge it to unblock other work (like #966).

**Pull request merged**
